rbac/management/role/definer.py

```diff
@@ -274,3 +283,117 @@ def delete_permission(permission: Permission):
+    """Create or update V2 role from V1 role during seeding."""
+    try:
+        v2_role, v2_created = SeededRoleV2.objects.update_or_create(
+            uuid=v1_role.uuid,
+            defaults={
+                "name": display_name,
+                "description": description,
+                "tenant": public_tenant,
+                "v1_source": v1_role,
+            },
+        )
+
+        if v2_created:
+            logger.info("Created V2 system role %s.", display_name)
+        else:
+            logger.info("Updated V2 system role %s.", display_name)
+
+        v2_role.permissions.clear()
+        v1_permissions = [access.permission for access in v1_role.access.all()]
+        if v1_permissions:
+            v2_role.permissions.set(v1_permissions)
+            logger.info("Added %d permissions to V2 role %s.", len(v1_permissions), display_name)
+
+        resource_service = ImplicitResourceService.from_settings()
+        scope = resource_service.scope_for_role(v1_role)
+
+        platform_role = platform_roles[(DefaultAccessType.USER, scope)]
+        if v1_role.platform_default:
+            platform_role.children.add(v2_role)
+            logger.info("Added %s as child of platform role %s", display_name, platform_role.name)
+        else:
+            platform_role.children.discard(v2_role)
+
+        admin_platform_role = platform_roles[(DefaultAccessType.ADMIN, scope)]
+        if v1_role.admin_default:
+            admin_platform_role.children.add(v2_role)
+            logger.info("Added %s as child of admin platform role %s", display_name, admin_platform_role.name)
+        else:
+            admin_platform_role.children.discard(v2_role)
+        return v2_role
+    except Exception as e:
+        logger.error(f"Failed to seed V2 role for {display_name}: {e}")
+        return None
+
+
+def _seed_platform_roles():
+    """Create the 6 platform roles (3 scopes × 2 access types)."""
+    public_tenant = Tenant.objects.get(tenant_name="public")
+    policy_service = GlobalPolicyIdService.shared()
+
+    platform_roles = {}
+
+    for access_type in DefaultAccessType:
+        for scope in Scope:
+            try:
+                platform_role = _create_single_platform_role(access_type, scope, policy_service, public_tenant)
+                platform_roles[(access_type, scope)] = platform_role
+            except DefaultGroupNotAvailableError:
+                logger.warning(
+                    "Default groups do not exist yet. Creating them now for %s %s scope",
+                    access_type.value,
+                    scope.name.lower(),
+                )
+                # Create the default groups
+                seed_group()
+
+                # Retry creating the platform role now that default groups exist
+                try:
+                    # Refresh policy service since groups were just created
+                    refreshed_policy_service = GlobalPolicyIdService.shared()
```

Comment:
We should probably just do `policy_service = GlobalPolicyIdService.shared()` here, so we don't try to re-use an instance that we know is invalid on the next iteration.